### DISCUSSION TITLE
Are refactors of rules accepted?

### DISCUSSION THREAD

**Discussion by OlaoluwaM** [eslint/eslint#15523] [Ideas]
Does eslint accept pull requests that refactor the code for existing rules?

**Comment by nzakas** [comment]
It depends on the reason for the refactor. If a refactor improves readability, reduces duplication, or improves performance, we will likely accept it; If the refactor only has aesthetic value, we likely will not. If you’re unsure, you can always open an issue describing what you’re thinking of doing to get feedback.
